src/overlays/actors/ovl_En_Daiku2/z_en_daiku2.c

```diff
@@ -28,60 +57,524 @@ const ActorInit En_Daiku2_InitVars = {
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE65B4.s")
+void EnDaiku2_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE66E4.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE6B40.s")
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gDPSetEnvColor(POLY_OPA_DISP++, 245, 155, 0, 255);
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount, NULL,
+                     EnDaiku2_PostLimbDraw, &this->actor);
+    func_80BE7718(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE6BC0.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE6CFC.s")
+void func_80BE7504(EnDaiku2* this, Vec3f* arg1, Vec3f* arg2, Vec3f* arg3, f32 arg4, s16 arg5) {
+    s16 i;
+    EnDaiku2Particle* particle = &this->particles[0];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE6D40.s")
+    for (i = 0; i < ARRAY_COUNT(this->particles); i++, particle++) {
+        if (!particle->isEnabled) {
+            particle->isEnabled = true;
+            particle->unk_04 = *arg1;
+            particle->unk_10 = *arg2;
+            particle->unk_1C = *arg3;
+            particle->unk_30 = arg4;
+            particle->unk_34 = arg5;
+            particle->unk_28.x = randPlusMinusPoint5Scaled(30000.0f);
+            particle->unk_28.y = randPlusMinusPoint5Scaled(30000.0f);
+            particle->unk_28.z = randPlusMinusPoint5Scaled(30000.0f);
+            break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE6EB0.s")
+void func_80BE7600(EnDaiku2* this, GlobalContext* globalCtx) {
+    s32 i;
+    EnDaiku2Particle* particle = &this->particles[0];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE6EF0.s")
+    for (i = 0; i < ARRAY_COUNT(this->particles); i++, particle++) {
+        if (particle->isEnabled) {
+            particle->unk_04.x += particle->unk_10.x;
+            particle->unk_28.x += 0x100;
+            particle->unk_28.z += 0x130;
+            particle->unk_04.y += particle->unk_10.y;
+            particle->unk_04.z += particle->unk_10.z;
+            particle->unk_10.y += particle->unk_1C.y;
+            if (particle->unk_34 != 0) {
+                particle->unk_34 -= 1;
+            } else {
+                particle->isEnabled = false;
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Daiku2/func_80BE71A0.s")
+void func_80BE7718(EnDaiku2* this, GlobalContext* globalCtx) {
+    s32 i;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    EnDaiku2Particle* particle = &this->particles[0];
+    s32 object;
```

Comment:
```suggestion
    s32 objectId;
```
I think I'd call this like this or even `objectIdx` or `objectIndex`